docs/DeveloperGuide.md

```diff
@@ -1,38 +1,235 @@
+
+Step 1. As seen from the sequence diagram above, when the AddMealCommand is executed via the `execute` method in 
+LifeTracker, the user's input is first parsed to determine how he/she wants to input it. Either method sets the food, 
+date and meal type features necessary to create a new meal.
+
+Step 2. The constructor for the `Meal` class is called which instantiates a new instance of Meal using the
+parameters provided.
+
+Step 3. `mealStorage` saves the meal to the database and then `ui` prints out the confirmation of the meal added.
+
+## [Proposed] Delete meal feature
+
+### Proposed Implementation
+
+In order to delete a meal from their meal history, a user has to type `delete [MEAL_INDEX]` into the CLI. Users can see 
+which meal they want to delete by viewing the mealData.csv file.
 
-{Describe the target user profile}
+<img src="uml/DeleteMealCommand1.png" alt="Sequence Diagram" width="500">
 
-### Value proposition
+In the above implementation, DeleteMealCommand parses the user input to obtain the index to delete and proceeds to 
+delete it via the method from mealStorage() and prints out the deleted meal to the user.
 
-{Describe the value proposition: what problem does it solve?}
+## [Proposed] View feature
 
-## User Stories
+### Proposed Implementation
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The proposed view mechanism is facilitated by `ViewUserCommand`. It extends `Command` and overrides the 
+`execute` method in the `Command` class. 
+
+It stores the user's data internally as `user` and the meals consumed by the user as `meals`. It also 
+initializes the UI for calories as `calorieUI`.
+
+Given below is an example usage scenario and how the view feature behaves at each step.
+
+Step 1. The user launches the application and calls the `view` command. The `ViewUserCommand` will be initialized
+with the current user and meal storage state. `user` and `meals` will point to the user storage and meal state 
+respectively.
+
+<img src="uml/ViewUserCommandSD1.png" alt="Sequence Diagram 1" width="500">
+
+
+The user will then be presented with a menu showing the different user details he can view as seen in the code
+snippet below.  
+
+````
+View user settings
+1. View Name
+2. View Weight
+3. View Height
+4. View Age      
+5. View Gender
+6. View Daily Caloric limit
+7. View Calories left today
+8. Back
+````
+
+Step 2. The user chooses to view his weight by inputting the number `2` to choose `View Weight`.
+This calls the getter method `getWeight()` in the entity `User` to return the current weight of the user
+and initializes the variable `weight` with that value. 
+
+<img src="uml/ViewUserCommandSD2.png" alt="Sequence Diagram 2" width="500">
```

Comment:
The label for the "alt" frame should have the diagonal "folded in" bottom right corner (same for opt and par frames)